$NetBSD$

Adds support for BSD based builds.

1. PREFIX path is configured via BSD_PREFIX_PATH for BSDs.

2. Boost include directory is set from bsd.cmake for BSDs.

--- cmake/CMakeLists.txt.orig	2019-08-18 00:42:17.620253273 +0000
+++ cmake/CMakeLists.txt
@@ -20,12 +20,22 @@ if(TARGET)
         message("CMake files for ${TARGET} not found, using ${DCMAKE}")
         include(${DCMAKE})
     endif()
+    if(${TARGET} STREQUAL "bsd")
+        if(NOT DEFINED BSD_PREFIX_PATH)
+            message(FATAL_ERROR "Warning: BSD_PREFIX_PATH Variable Not Defined!")
+        else()
+            message("Setting ${BSD_PREFIX_PATH} as the prefix path.")
+        endif()
+    endif()
 else()
     message("TARGET not passed, using ${DCMAKE}")
     include(${DCMAKE})
 endif(TARGET)
 
-set(BOOST_INCLUDEDIR ${lib_base}/boost_1_54_0)
+if(NOT(${TARGET} STREQUAL "bsd"))
+    set(BOOST_INCLUDEDIR ${lib_base}/boost_1_54_0)
+endif()
+
 find_package(Boost REQUIRED)
 
 # Include
